### ISSUE TITLE
Ghosts'n Goblins coin button works randomly

### ISSUE THREAD

**Issue by FeRcHuLeS** [libretro/mame2003-plus-libretro#746]
Pressing the coin button works sometimes and the only way to make it working is pressing the coin button repeatedly until a credit pop up, It is not related to the controller since I was playing another titles in this core neither a exclusive configuration for this game.

**Comment by arcadez2003** [comment]
You didn't mention what platform your using and or what frontend, does the game do this
in mame2003.?? also handy hint i used to get some games like that with FBN on the xbox
try a quick tap when adding a coin rather than a longer press i found that would help.

**Comment by FeRcHuLeS** [comment]
The platform is Windows x86 on Retroarch, It is the same as mame 2003, I figured out this happens because of wrong coding of instering a coin, while another games pop up a credit after thThe platform is Windows x86 on Retroarch, It is the same as mame 2003, I figured out this happens because of wrong coding of instering a coin, while another games pop up a credit after the coin button is released this game needs to press and release the button as fast as possible as you suggested.e coin button is released this game needs to press and release the button as fast as possible as you suggested.

**Comment by Wilstorm** [comment]
I tested this on a Raspberry Pi 3 with m3plus build aa8e74a and 2003 build aa607d0 using Retroarch 1.7.6 and it couldn't find any issues. 

I tested inserting coins using the keyboard as well as a wireless XBOX 360 controller.

With the keyboard as long as I pressed and released the key at a normal speed it would register a credit. If I held it down for a few seconds and let up it wouldn't register a credit. It seems to register a credit on the release of the key if done within a certain time frame.

With the 360 controller it would register a credit on the down press of the button whether it was slow or fast it didn't make a difference and worked perfectly adding rapid credits or slowly. Basically it's registering on the down press regardless of the release time.

It ran around 60 FPS consistently. I tested both the World and US versions (gng, gnga & gngt) and all seemed to work fine. It must be platform specific?
